snowshu/core/graph.py

```diff
@@ -365,8 +367,26 @@ def _process_downstream_relation_set(
                 f'View dependencies are not allowed by SnowShu.'
             )
 
+        catalog_dict = dict(map(lambda x: (x.dot_notation, dict(map(lambda a: (a.name, a.data_type), x.attributes))),
+                                full_relation_set))
+
         for downstream_relation in downstream_set:
             for upstream_relation in upstream_without_downstream:
+                # validate whether relation is valid
+                attributes = dict(filter(lambda item: item[0] in ['remote_attribute', 'local_attribute'],
+                                         relationship['edge_attributes'].items()))
+                remote_attribute = attributes.get('remote_attribute')
+                local_attribute = attributes.get('local_attribute')
+                is_remote_attribute_valid = remote_attribute in catalog_dict.get(upstream_relation.dot_notation, {})
+                is_local_attribute_valid = local_attribute in catalog_dict.get(downstream_relation.dot_notation, {})
+
+                if not is_remote_attribute_valid or not is_local_attribute_valid:
```

Comment:
Maybe, I should accumulate all possible issues and only then run raise an issue?
cc @eculicny

Reply:
It would probably be best to do that. Then the user doesn't need to keep trying to run when there are multiple problems that can be identified on one validation.

Reply:
<img width="1258" alt="image" src="https://user-images.githubusercontent.com/98582037/171355410-6ab9e0f2-b463-4cc8-9345-e08a460384c5.png">

(i) It will accumulate relevant information for every single db